### PR TITLE
Internal messaging common encode

### PR DESCRIPTION
**Pull request by Kinju011**

### CODE REVIEW

File: ruby/neo4j/driver/internal/messaging/common/common_value_packer.rb [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+            end
+          end
+
+          def pack_internal_value(value)
+            case value.type_constructor
+            when DATE
+              pack_date(value.as_local_date)
+            when TIME
+              pack_time(value.as_offset_time)
+            when LOCAL_TIME
```
Comment (Kinju011) [inline ruby/neo4j/driver/internal/messaging/common/common_value_packer.rb:60]: Hello @klobuczek 
I have converted primitives data types into ruby but didn't invest much time in these types of data types,
but please let me know if needed to handle them too?

Comment (klobuczek) [reply ruby/neo4j/driver/internal/messaging/common/common_value_packer.rb:60]: yes, this part is good

File: ruby/neo4j/driver/internal/util/metadata_extractor.rb [hunk 2]
```diff
@@ -8,6 +8,150 @@ def initialize(result_available_after_metadata_key, result_consumed_after_metada
           @result_available_after_metadata_key = result_available_after_metadata_key
           @result_consumed_after_metadata_key = result_consumed_after_metadata_key
         end
+
+        def extract_query_keys(metadata)
+          keys_value = metadata['fields']
+          unless keys_value.nil? && keys_value.empty?
```
Comment (klobuczek) [inline ruby/neo4j/driver/internal/util/metadata_extractor.rb:14]: wrong

File: ruby/neo4j/driver/internal/util/metadata_extractor.rb [hunk 3]
```diff
@@ -8,6 +8,150 @@ def initialize(result_available_after_metadata_key, result_consumed_after_metada
...
+            # keys = Util::QueryKeys.new(keys_value.size)
```
Comment (klobuczek) [inline ruby/neo4j/driver/internal/util/metadata_extractor.rb:15]: QueryKeys merit conversion

**Review by klobuczek — APPROVED** [review]

**Pull request merged**